### PR TITLE
feat(validator): handle the output when L1 reorg occurs

### PR DESCRIPTION
**Pull request by 0xHansLee**
# Description

Related to https://github.com/chainlight-io/2023-kroma-network-audit-issues-client/issues/11 reported by theori audit.
L1 reorg was not taken into account in the challenger before.
If L1 reorg occurs, the challenger verifies the newly generated output checkpoint.

### CODE REVIEW

File: components/validator/challenger.go [hunk 1]
```diff
@@ -283,10 +283,16 @@ func (c *Challenger) subscribeL2OutputSubmitted() {
 		select {
 		case ev := <-c.l2OutputSubmittedEventChan:
 			c.log.Info("watched output submitted event", "l2BlockNumber", ev.L2BlockNumber, "outputRoot", ev.OutputRoot, "outputIndex", ev.L2OutputIndex)
-			// validate all outputs between the checkpoint and the current outputIndex
-			for i := new(big.Int).Add(c.checkpoint, common.Big1); i.Cmp(ev.L2OutputIndex) != 1; i.Add(i, common.Big1) {
+			// if the emitted output index is less than or equal to the checkpoint, it is considered reorg occurred.
+			if ev.L2OutputIndex.Cmp(c.checkpoint) <= 0 {
 				c.wg.Add(1)
-				go c.handleOutput(new(big.Int).Set(i))
+				go c.handleOutput(ev.L2OutputIndex)
```
Comment (kangsorang) [inline components/validator/challenger.go:289]: This doesn't seem to be a problem here, but it's probably a good idea not to put the pointer directly into the go routine.
```suggestion
				go c.handleOutput(new(big.Int).Set(ev.L2OutputIndex))
```

Comment (0xHansLee) [reply components/validator/challenger.go]: Oh, I made the same mistake. TY!

**Comment by 0xHansLee** [comment]
For future work, it would be good to add some E2E challenge tests for L1 reorg situation.

**Review by seolaoh — APPROVED** [review]
LGTM

**Review by kangsorang — APPROVED** [review]
LGTM

**Review by Pangssu — APPROVED** [review]
LGTM

**Pull request merged**